spec/models/plant_accession_spec.rb

```diff
@@ -47,6 +47,25 @@
     end
   end
 
+  context 'unique attributes' do
+    it 'disallows duplicate plant_accession and originating_organisation' do
```

Comment:
Yes, and the new index should be tested with an update_column call (i.e. bypassing the AR layer). While it is difficult to test a race condition, this should be a good-enough equivalent.